UserBackgrounds/index.tsx

```diff
@@ -119,12 +134,13 @@ export default class UserBackgrounds extends BasePlugin {
             children: React.ReactElement;
         };
 
-        function BannerContainer({user, bannerType, children}: params) {
+        function BannerContainer({user, guildId, bannerType, children}: params) {
+            const guildMember = DiscordModules.GuildMemberStore.getMember(guildId, user.id);
             const banner: banner = useStateFromStores([BannerStore], () => BannerStore.getBanner(user.id), null, _.isEqual);
-            const [selection, setSelection] = useState(banner == null ? 1 : 0);
+            const [selection, setSelection] = useState(guildMember?.banner == null ? user.banner == null ? 2 : 1 : 0 );
             const ref = useRef(null);
-            const currentBanner = useMemo(() => (selection === 1 || banner == null) ? getBannerURL(user, true) : banner?.background, [banner, user, selection]);
-            const currentOrientation = useMemo(() => (banner != null && selection === 0) ? banner.orientation : void 0, [banner, selection]);
+            const currentBanner = useMemo(() => ((selection === 0 || banner == null) && guildMember?.banner) ? getGuildBannerURL(guildMember, guildId, true) : (selection === 1 || banner == null) ? getBannerURL(user, true) : banner?.background, [banner, user, selection]);

```

Comment:
This could use some cleanup and less ternaries.

Reply:
Tbh, idk how i could clean it up. maybe because its late or im just dumb

Reply:
Convert it to use normal `if() {} else {}` structure.